### PR TITLE
Update modules with 4.x DevEx deprecations

### PR DESCRIPTION
**Pull request by bmcelvee**
@adambkaplan I'm going through the deprecations doc looking for existing modules to figure out where updates need to be made. I've flagged a few so far. If you have time, I'd like to sync to make sure these all get updated appropriately. 

Thanks!

### CODE REVIEW

**Comment by bmcelvee** [comment]
I also commented out the force pull modules. Should those be removed entirely?

File: modules/builds-automatically-add-source-clone-secrets.adoc [hunk 1]
```diff
@@ -15,6 +15,8 @@ automatically use the credentials stored in the referenced `Secret` to
 authenticate to a remote Git repository, without requiring further
 configuration.
 
+//Build overrides - annotations needs update?
```
Comment (adambkaplan) [inline modules/builds-automatically-add-source-clone-secrets.adoc]: Nope

File: modules/builds-disabling-build-strategy-globally.adoc [hunk 2]
```diff
@@ -7,6 +7,8 @@
 [id="builds-disabling-build-strategy-globally-{context}"]
 = Disabling access to a build strategy globally
 
+//Annotations - needs update?
```
Comment (adambkaplan) [inline modules/builds-disabling-build-strategy-globally.adoc]: Nope

File: modules/builds-service-serving-certificate-secrets.adoc [hunk 3]
```diff
@@ -11,6 +11,8 @@ masters.
 
 .Procedure
 
+//Annotations - needs update?
```
Comment (adambkaplan) [inline modules/builds-service-serving-certificate-secrets.adoc]: Nope

File: modules/builds-troubleshooting-service-certificate-generation.adoc [hunk 4]
```diff
@@ -7,6 +7,8 @@
 
 If your request for access to resources is denied:
 
+//Annotations - remove?
```
Comment (adambkaplan) [inline modules/builds-troubleshooting-service-certificate-generation.adoc]: Keep - this is not related to the default/override annotations used in the cluster config.

File: modules/builds-use-custom-builder-image.adoc [hunk 5]
```diff
@@ -17,6 +17,8 @@ your custom builder image to execute your custom build logic.
 
 .Procedure
 
+//Annotations - needs update?
+
```
Comment (adambkaplan) [inline modules/builds-use-custom-builder-image.adoc]: No.

It feels like there is an advanced module for custom strategy where we talk about exposing the Docker socket. That needs a real deprecation notice.

**Comment by bmcelvee** [comment]
Closing PR. Will address additional required changes in a new PR.
